packages/backend/src/apps/formsg/common/get-data-out-metadata.ts

```diff
@@ -339,35 +353,35 @@ async function getDataOutMetadata(
   })
 
   /**
-   * This is a hack to match the question number to the form as closely as possible.
-   * In formsg, the headers are not numbered, so we need to exclude them from the question number.
-   * But we also need to keep track of the headers between questions, so we can order them correctly.
-   * The regenerated order will be like so:
-   * Example given form:
-   * Header 0.9991
-   * Header 0.9992
-   * Question 1
-   * Question 2
-   * Sub Heading 2.9991
-   * Question 3
-   * Header 3.9991
-   * Sub Heading 3.9992
-   * Question 4
-   * Sub Heading 4.9991
-   * Question 4
+   * We ignore all 'image' and 'section' (aka headers) field types
+   * Paragraphs are not returned in encryptedContent
    */
   let questionOrder = 0
-  let headerOrderBetweenQuestions = 0
   for (const [fieldId, fieldData] of fields) {
-    if (fieldData.fieldType !== 'section') {
-      // reset order between questions (altho not necessary)
-      headerOrderBetweenQuestions = 0
-      questionOrder++
-      fieldData.order = questionOrder
-    } else {
-      headerOrderBetweenQuestions++
-      fieldData.order = questionOrder + +`0.999${headerOrderBetweenQuestions}`
+    // ignore image fields, dont even increment question order
+    if (fieldData.fieldType === 'image' || fieldData.fieldType === 'section') {
+      fieldMetadata[fieldId] = { isHidden: true }
+      continue
     }
+
+    // increment question order for each field
+    questionOrder++
+    fieldData.order = questionOrder
+
+    // if this is an MRF step, we only show the fields that are editable
+    if (questionIdsToShowForMrf && !questionIdsToShowForMrf.has(fieldId)) {
+      fieldMetadata[fieldId] = {
+        question: { isHidden: true },
+        answer: { isHidden: true },
```

Comment:
```suggestion
        answer: { isHidden: true },
        answerArray: { isHidden: true },
```

think need to add for `answerArray` as well to handle checkboxes and tables